src/lightning/data/utilities/broadcast.py

```diff
@@ -0,0 +1,187 @@
+            # are going to be alive for a very long time (~ 4 days) but retries every 120 seconds
+            total=_CONNECTION_RETRY_TOTAL,
+            backoff_factor=_CONNECTION_RETRY_BACKOFF_FACTOR,
+            status_forcelist=[
+                408,  # Request Timeout
+                429,  # Too Many Requests
+                500,  # Internal Server Error
+                502,  # Bad Gateway
+                503,  # Service Unavailable
+                504,  # Gateway Timeout
+            ],
+        )
+        adapter = CustomRetryAdapter(max_retries=retry_strategy, timeout=_DEFAULT_REQUEST_TIMEOUT)
+        self.session = requests.Session()
+
+        self.session.hooks = {"response": _response}
+
+        if use_retry:
+            self.session.mount("http://", adapter)
+            self.session.mount("https://", adapter)
+
+        if auth_token:
+            self.session.headers.update({"Authorization": f"Bearer {auth_token}"})
+
+        self.log_function = log_callback or self.log_function  # type: ignore
+
+    @_http_method_logger_wrapper
+    def get(self, path: str) -> Any:
+        url = urljoin(self.base_url, path)
+        return self.session.get(url)
+
+    @_http_method_logger_wrapper
+    def post(
+        self, path: str, *, query_params: Optional[Dict] = None, data: Optional[bytes] = None, json: Any = None
+    ) -> Any:
+        url = urljoin(self.base_url, path)
+        return self.session.post(url, data=data, params=query_params, json=json)
+
+    @_http_method_logger_wrapper
+    def delete(self, path: str) -> Any:
+        url = urljoin(self.base_url, path)
+        return self.session.delete(url)
+
+    def log_function(self, message: str, *args: Any, **kwargs: Any) -> None:
+        """This function is used to log the messages in the client, it can be overridden by caller to customise the
+        logging logic.
+
+        We enabled customisation here instead of just using `logger.debug` because HTTP logging can be very noisy, but
+        it is crucial for finding bugs when we have them
+
+        """
+
+
+class ImmutableDistributedMap:
+    """The ImmutableDistributedMap enables to create a distributed key value pair in the cloud.
+
+    The first process to perform the set operation defines its value.
+
+    """
+
+    def __init__(self) -> None:
+        # Get the token
+        token = None
+        if os.getenv("LIGHTNING_CLOUD_URL") is not None:
+            payload = {"apiKey": os.getenv("LIGHTNING_API_KEY"), "username": os.getenv("LIGHTNING_USERNAME")}
+            url_login = os.getenv("LIGHTNING_CLOUD_URL", "") + "/v1/auth/login"
+            res = requests.post(url_login, data=json.dumps(payload))
+            if "token" not in res.json():
+                raise RuntimeError(
+                    f"You haven't properly setup your environment variables with {url_login} and data: \n{payload}"
+                )
+            token = res.json()["token"]
```

Comment:
This should probably be a function/method? Then you don't need a comment "get the token"

Reply:
sure.